Sources/Views/Cells/AudioMessageCell.swift

```diff
@@ -0,0 +1,171 @@
+/*
+ MIT License
+
+ Copyright (c) 2017-2018 MessageKit
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import UIKit
+import AVFoundation
+
+/// A subclass of `MessageContentCell` used to display video and audio messages.
+open class AudioMessageCell: MessageContentCell {
+
+    /// The `AudioCellDelegate` delegate
+    internal weak var audioCellDelegate: AudioCellDelegate?
+
+    /// The play button view to display on audio messages.
+    open lazy var playButton: UIButton = {
+        let playButton = UIButton.init(type: .custom)
+        let playImage = AudioMessageCell.getImageWithName("play")
+        let pauseImage = AudioMessageCell.getImageWithName("pause")
+        playButton.setImage(playImage?.withRenderingMode(.alwaysTemplate), for: .normal)
+        playButton.setImage(pauseImage?.withRenderingMode(.alwaysTemplate), for: .selected)
+
+        return playButton
+    }()
+
+    /// The time duration lable to display on audio messages.
+    open lazy var durationLabel: UILabel = {
+        let durationLabel = UILabel.init(frame: CGRect.zero)
+        durationLabel.textAlignment = .right
+        durationLabel.font = UIFont.systemFont(ofSize: 14)
+        durationLabel.text = "0:00"
+
+        return durationLabel
+    }()
+
+    open lazy var progressView: UIProgressView = {
+        let progressView = UIProgressView.init(progressViewStyle: .default)
+        progressView.progress = 0.0
+
+        return progressView
+    }()
+
+    // MARK: - Internal Methods
+    internal class func durationString(from duration: Float) -> String {
```

Comment:
Not sure a `class` func is the right way to approach this, is there a better way? At the very least this should be `open`

Reply:
@nathantannar4 you are right, this class method is not the best approach. I had removed it from AudioMessageCell and instead I created a new delegate methods on MessageDisplayDelegate. In this case user can proide it's own implementation for formatting the audio sound duration or can use the default implementation.